src/frontend/hooks/goose-form.js

```diff
@@ -96,55 +108,66 @@ const useGooseForm = () => {
 
         const res = await api.formUpload(data);
 
-        imageUrls = res.data.data;
-        if (imageUrls.length === 0) {
+        imageUrls2 = res.data.data;
+
+        if (imageUrls2.length === 0) {
           throw new Error('Failed to upload image.');
         }
       }
-
+      }
+    
       const gooseInfo = {
         name: gooseName,
         description,
         updatedAt: Date.now(),
       };
-
-      if (gooseName && description && imageUrls.length > 0) {
+      
+      if (gooseName && description && imageUrls2.length > 0) {
         if (params.gooseId) {
           await api.geeseInfo.updateGeeseInfo(params.gooseId, gooseInfo);
           await Promise.all(
             imagesToDelete.map((imageId) => {
               return api.geeseInfo.deleteGeeseImages(imageId);
             }),
           );
+
+          // upload images if new files are added
+          if (files.length >0){
           await api.geeseInfo.addGeeseImages(
-            imageUrls.map((image) => {
+            imageUrls2.map((image) => {
               return {
                 gooseId: params.gooseId,
                 imgDir: image,
               };
             }),
           );
-        } else {
+          }
+        } 
+        else {
           // // TODO when new goose is implemented
           const res = await api.geeseInfo.addGeeseInfo(gooseInfo);
           await api.geeseInfo.addGeeseImages(
-            imageUrls.map((image) => {
+            imageUrls2.map((image) => {
               return {
                 gooseId: res.data[0],
                 imgDir: image,
               };
             }),
           );
         }
-      } else {
+       
+      } 
+
+
+      else {
         throw new Error('Please fill all the required fields.');
       }
-
       // onSuccess:
       closeForm();
     } catch (e) {
       // TODO: Display "could not add/update" error to user as dialogue.
       // eslint-disable-next-line no-console
+      
       console.error(e);
     }
   }, [images, params, gooseName, description, closeForm, imagesToDelete]);
```

Comment:
One interesting bug that I noticed was that the form wouldn't save when you changed nothing. This is because `imageUrls` isn't in this dependency array and since `useCallback` is employed, the initial value of the state (empty array) is used and isn't recomputed because `imageUrls` isn't in this array. It be should be added